Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -68,9 +71,26 @@ async function run() {
         webDeployPkg = availableWebPackages[0];
 
         var isFolderBasedDeployment = utility.isInputPkgIsFolder(webDeployPkg);
+        var generateTempFolder = JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution || addWebConfig;
 
+        if(generateTempFolder) {
+            var folderPath = await fileTransformationsUtility.generateTemporaryFolder(isFolderBasedDeployment, webDeployPkg);
+        }
+        
         if(JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution) {
-            var output = await fileTransformationsUtility.fileTransformations(isFolderBasedDeployment, JSONFiles, xmlTransformation, xmlVariableSubstitution, webDeployPkg);
+            await fileTransformationsUtility.fileTransformations(isFolderBasedDeployment, JSONFiles, xmlTransformation, xmlVariableSubstitution, folderPath);
+        }
+
+        if (addWebConfig) {
+            switch (appType.toLowerCase()) {
+                case "node":
+                    addWebConfigForNode(folderPath, webConfigParameters);
+                    break;
+            }
+        }
+
+        if(generateTempFolder) {
+            var output = await fileTransformationsUtility.archiveFolder(isFolderBasedDeployment, webDeployPkg, folderPath);
```

Comment:
Can you try to simplify the above logic? For example, we shouldnt need "JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution" check multiple times. And if generateTempFolder is false, the next 2 checks will always be false..